ae/core/autogen_wrapper.py

```diff
@@ -237,21 +243,30 @@ def is_planner_termination_message(x: dict[str, str])->bool: # type: ignore
                 try:
                     content_json = parse_response(content)
                     _terminate = content_json.get('terminate', "no")
+                    final_response = content_json.get('final_response', None)
                     if(_terminate == "yes"):
                         should_terminate = True
+                        if final_response:
+                            print("Notifying Final Answer")
+                            notify_planner_messages(final_response, level="answer")
                 except json.JSONDecodeError:
                     print("Error decoding JSON content")
                     should_terminate = True
             
              return should_terminate # type: ignore
         
-        task_delegate_agent = autogen.ConversableAgent(
+        task_delegate_agent = UserProxyAgent_SequentialFunctionExecution(
             name="user",
             llm_config=False, 
             system_message=LLM_PROMPTS["USER_AGENT_PROMPT"],
             is_termination_msg=is_planner_termination_message, # type: ignore
             human_input_mode="NEVER",
             max_consecutive_auto_reply=self.number_of_rounds,
+            code_execution_config={
+                                "last_n_messages": 1,
```

Comment:
What's the significance of setting this to 1? Maybe some inline comment here for the reasoning behind it?

Reply:
Dont need it, removed code_execution_config